nmostesting/mocks/Registry.py

```diff
@@ -271,3 +417,198 @@ def heartbeat(version, node_id):
         return jsonify({"health": int(time.time())})
     else:
         abort(404)
+
+
+@REGISTRY_API.route('/x-nmos/query', methods=["GET"], strict_slashes=False)
+def query_root():
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['v1.0/', 'v1.1/', 'v1.2/', 'v1.3/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>', methods=["GET"], strict_slashes=False)
+def query(version):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['devices/', 'flows/', 'nodes/', 'receivers/', 'senders/', 'sources/', 'subscriptions/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+def compare_resources(resource1, resource2):
+    try:
+        return NMOSUtils.compare_resource_version(resource1['version'], resource2['version'])
+    except Exception as e:
+        print(e)
+        return 0
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/<resource>', methods=["GET"], strict_slashes=False)
+def query_resource(version, resource):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    MIN_SINCE = "0:0"
+    MAX_UNTIL = NMOSUtils.get_TAI_time()
+
+    base_data = []
+
+    if request.args.get('paging.since') or request.args.get('paging.until'):
+        registry.pagination_used = True
+
+    since = request.args.get('paging.since') if request.args.get('paging.since') else MIN_SINCE
+    until = request.args.get('paging.until') if request.args.get('paging.until') else MAX_UNTIL
+    new_until = until
```

Comment:
I think this needs to handle a `paging.limit` query parameter too, right?
Something like:
```py
limit = request.args.get('paging.limit') if request.args.get('paging.limit') else registry.paging_limit
```

Hmm, actually, can't they all be written more elegantly like:
```py
since = request.args.get('paging.since') or MIN_SINCE
```

I haven't checked precisely what `registry.pagination_used = True` is tracking, so I don't know if that needs to take `paging.limit` into account.

Reply:
Yeah, registry.pagination_used = True is used by Test_03 of IS0404 to whether the Query API is being exercised, and if it is using the REST API that pagination is being used.  However, Test_03 seems a little redundant, in that Test_04 and Test_05 explicitly check that Senders and Receivers can be browsed, and each of these implicitly checks pagination (i.e. if pagination not implemented in Controller, you won't see all the Senders/Receivers and will fail the test).

Reply:
However, the use of `registry.pagination_used` attempted to explicitly check for pagination use by the Controller

Reply:
You're one adrift, the pagination checks are in `test_03` and `test_04`. It's `test_02` to check the Query API is exercised, and that checks that either a websocket subscription has been set up or the query api was called

Reply:
Good catch @hrbriggs!

Reply:
Also @garethsb you're absolutely right that all these can be more elegantly expressed using the `or` operator instead of testing for truthiness. Will make those changes.

Reply:
Done